version: '3.7'

services:
  # PHP Application
  app:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: emergency_call_app
    restart: unless-stopped
    working_dir: /var/www/html
    volumes:
      - ./:/var/www/html
      - ./docker/php/local.ini:/usr/local/etc/php/conf.d/local.ini
    networks:
      - emergency_call_network

  # Nginx Web Server
  nginx:
    image: nginx:alpine
    container_name: emergency_call_nginx
    restart: unless-stopped
    ports:
      - "8080:80"
    volumes:
      - ./:/var/www/html
      - ./docker/nginx/conf.d:/etc/nginx/conf.d
    depends_on:
      - app
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.emergency-call.entrypoints=websecure"
      - "traefik.http.routers.emergency-call.rule=Host(`emergency.numanoglu.org`)"
      - "traefik.http.routers.emergency-call.tls=true"
      - "traefik.http.routers.emergency-call.tls.certresolver=myresolver"
      - "traefik.docker.network=traefik_web"
    networks:
      - emergency_call_network
      - traefik_web

  # MySQL Database
  db:
    image: mysql:8.0
    container_name: emergency_call_db
    restart: unless-stopped
    environment:
      MYSQL_DATABASE: emergency_call
      MYSQL_ROOT_PASSWORD: rootpassword
      MYSQL_USER: emergency_user
      MYSQL_PASSWORD: emergency_password
    ports:
      - "3306:3306"
    volumes:
      - dbdata:/var/lib/mysql
      - ./docker/mysql/init:/docker-entrypoint-initdb.d
    networks:
      - emergency_call_network

networks:
  emergency_call_network:
    driver: bridge
  traefik_web:
    external: true    
volumes:
  dbdata: 